OpenLIFUData/OpenLIFUData.py

```diff
@@ -1191,15 +1199,22 @@ def load_session(self, subject_id, session_id) -> None:
         self.getParameterNode().loaded_session = new_session
 
     def _on_transducer_transform_modified(self, transducer: SlicerOpenLIFUTransducer) -> None:
+        session = self.getParameterNode().loaded_session
+
+        if session is None:
+            return
+
+        # Revoke transducer tracking approval if there was any
+        if session.transducer_tracking_is_approved():
+            session.toggle_transducer_tracking_approval() # revoke approval
+            self.getParameterNode().loaded_session = session # remember to write the updated session object into the parameter node
+
         # Revoke any possible virtual fit approval if the transducer whose transform was just modified
         # belongs to an active session
-        session = self.getParameterNode().loaded_session
         if (
-            session is None
-            or session.session.session.virtual_fit_approval_for_target_id is None
+            session.session.session.virtual_fit_approval_for_target_id is not None
+            and session.get_transducer_id() == transducer.transducer.transducer.id
         ):
-            return
-        if session.get_transducer_id() == transducer.transducer.transducer.id:
             session.approve_virtual_fit_for_target(None) # revoke approval
```

Comment:
I think it would help to include a error/warning message box here to let the user know that transducer tracking approval is being revoked due to transform modification? Otherwise, the status changes without the user necessarily knowing when or why it happens.

Reply:
agreed, ty